### ISSUE TITLE
Performance verification: Reproduction results on Visdrone dataset

### ISSUE THREAD

**Issue by cjchnws** [wxz0426/RSVDet#2]
**Reproduction Results:**
EarlyStopping: Training stopped early as no improvement observed in last 100 epochs. Best results observed at epoch 528, best model saved as best.pt.
To update EarlyStopping(patience=100) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.

628 epochs completed in 123.205 hours.
Optimizer stripped from runs\detect\train\weights\last.pt, 4.1MB
Optimizer stripped from runs\detect\train\weights\best.pt, 4.1MB

Validating runs\detect\train\weights\best.pt...
Ultralytics YOLOv8.2.22 🚀 Python-3.10.15 torch-2.5.1+cu118 CUDA:0 (NVIDIA GeForce RTX 4090, 24564MiB)
RSVDet summary (fused): 210 layers, 1850645 parameters, 0 gradients

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:14<00:00,  1.29it/s]
                   all        548      38759      0.583      0.459      0.492      0.303
            pedestrian        548       8844      0.642       0.49       0.55      0.272
                people        548       5125      0.598      0.429      0.468        0.2
               bicycle        548       1287      0.317      0.276      0.237      0.112
                   car        548      14064      0.773      0.831      0.857      0.626
                   van        548       1975      0.633      0.499      0.537      0.388
                 truck        548        750      0.534      0.408      0.426      0.297
              tricycle        548       1045       0.55      0.358      0.384      0.229
       awning-tricycle        548        532      0.404      0.194      0.236      0.155
                   bus        548        251      0.756      0.562       0.65       0.47
                 motor        548       4886      0.629      0.548      0.576      0.281
Speed: 0.1ms preprocess, 6.8ms inference, 0.0ms loss, 1.9ms postprocess per image
Results saved to runs\detect\train
**Comments:**
Here are the results of my reproduction: mAP_50 is 49.2 and mAP50-95 is 30.3. Although they are slightly lower than the results reported in the paper (50.1 and 31.0), I still believe this is an excellent piece of work and very much worth studying.
Thanks again for your contribution!
**Question**
I'm curious about the training efficiency. The training process seems significantly slower than expected. Could you provide some insights on why this might be the case?
Environment Details:
OS: Windows
Python: 3.10.15
PyTorch: 2.5.1+cu118
GPU: NVIDIA GeForce RTX 4090 (24GB)
RAM: 128GB

**Comment by wxz0426** [comment]
Thank you very much for your reproduction. I believe the main reason for the slow training speed is the extensive use of custom convolution kernels in the model. These custom convolution kernels introduce additional computational overhead during the training process, which in turn significantly slows down the training progress.
Regarding the optimization of inference speed, there is a very specific and effective method: place the custom convolution kernels in the model initialization (__init__) to define them only once, instead of recreating them in each forward propagation.

The above are my views on the question you raised. We also sincerely hope that our work can help you and look forward to your citation.
